konig-schemagen/src/main/java/io/konig/schemagen/sql/RdbmsShapeGenerator.java

```diff
@@ -232,10 +230,26 @@ private PropertyConstraint createSyntheticKey(Shape rdbmsShape,String suffix, UR
 		
 		pc.setMaxCount(1);
 		pc.setMinCount(1);
-		pc.setStereotype(Konig.syntheticKey);
+		if("_PK".equals(suffix)){
```

Comment:
Again, why are we testing that "_PK" equals the suffix. This seems like a really bad idea.  Perhaps there should be some additional information passed as a parameter to control this switch.

Reply:
This check is added in order to avoid foreign keys (suffix ="_FK") to get AUTO_INCREMENT (being added in case of konig:syntheticKey stereotype) appended to them in the generated SQL.  I guess we can add something like isPrimaryKey instead of this condition.